pkgs/servers/http/couchdb/2.0.0.nix

```diff
@@ -0,0 +1,53 @@
+{ stdenv, fetchurl, erlang, icu, openssl, spidermonkey
+, coreutils, bash, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  name = "couchdb-${version}";
+  version = "2.0.0";
+
+  src = fetchurl {
+    url = "mirror://apache/couchdb/source/${version}/apache-${name}.tar.gz";
+    sha256 = "1jkfx6g9anrgmkhrkcn50axcamragranwsciw1rhmi86rglkrbyc";
+  };
+
+  buildInputs = [ erlang icu openssl spidermonkey makeWrapper ];
+
+  patches = [ ./jsapi.patch ];
+  postPatch = ''
+    substituteInPlace src/couch/rebar.config.script --replace '-DHAVE_CURL -I/usr/local/include' "-DHAVE_CURL -I/usr/local/include $NIX_CFLAGS_COMPILE"
+
+    patch bin/rebar <<EOF
```

Comment:
It does not work.  "bin/rebar" is basically a bytecode archive  file with a shebang in front to call the interpreter.  For some reason that I could not debug, using substituteInPlace does not work in the same way as using patch, and causes the resulting file to be not executable correctly.  For that reason, I kept it that way.